### PR TITLE
[GraphBolt] Fix `blocks` in minibatch when facing with empty edges in subgraph.

### PR DESCRIPTION
**Pull request by yxy235**
## Description

related to #7410 

TODO: 
1. add test case: get blocks from homo graph minibatch
2. add test case: check if node type of `seeds` still occurs in src_node_type of edges, it will always be passed down.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: tests/python/pytorch/graphbolt/test_minibatch.py [hunk 1]
```diff
@@ -336,6 +336,76 @@ def test_get_dgl_blocks_hetero():
...
+            reverse_relation: gb.CSCFormatBase(
+                indptr=torch.tensor([0, 0, 0, 1, 2]),
+                indices=torch.tensor([1, 0]),
+            ),
+        },
+        {
+            relation: gb.CSCFormatBase(
+                indptr=torch.tensor([0, 1, 2]), indices=torch.tensor([1, 0])
+            ),
+            reverse_relation: gb.CSCFormatBase(
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/test_minibatch.py:355]: so `FusedCSCSamplingGraph.sample_neighbors()` always returns sampled_csc for each edge type even it's empty? Could we mention this in docstring or documentation? probably in `sample_neighbors()`?

Comment (yxy235) [reply tests/python/pytorch/graphbolt/test_minibatch.py:355]: Yes. Let me add docstring.

**Comment by Rhett-Ying** [comment]
@Skeleton003 please work on this PR. the code change in python core is done. you just need to add more testcases, just cover all possible scenarios we may hit. Please refer to the `description` for the issue this PR is targeting on.

TODO:

    add test case: get blocks from homo graph minibatch
    add test case: check if node type of seeds still occurs in src_node_type of edges, it will always be passed down.

File: python/dgl/graphbolt/minibatch.py [hunk 2]
```diff
@@ -208,15 +223,21 @@ def cast_to_minimum_dtype(v: CSCFormatBase):
...
-                }
                 num_src_nodes = {
-                    ntype: nodes.size(0)
-                    for ntype, nodes in original_row_node_ids.items()
+                    ntype: (
+                        original_row_node_ids[ntype].size(0)
+                        if original_row_node_ids.get(ntype) is not None
+                        else 0
+                    )
+                    for ntype in set(node_types)
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/minibatch.py:232]: seems `node_types` should be a set instead list, why not directly instantiated as set at the beginning?

File: tests/python/pytorch/graphbolt/test_minibatch.py [hunk 3]
```diff
@@ -328,14 +338,96 @@ def test_get_dgl_blocks_hetero():
...
+        {
+            ("n1", "e1", "n1"): ([0, 1, 1], [1, 2, 0]),
+            ("n1", "e2", "n2"): ([0, 1, 2], [1, 0, 2]),
+        }
+    )
+
+    gb_g = gb.from_dglgraph(hg, is_homogeneous=False)
+
+    train_set = gb.HeteroItemSet(
+        {"n1:e1:n2": gb.ItemSet(torch.LongTensor([[0, 1]]), names="seeds")}
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/test_minibatch.py:360]: "n1:e1:n2" This canonical etype does not exist. should be `n1:d2:n2`?

File: tests/python/pytorch/graphbolt/test_minibatch.py [hunk 4]
```diff
@@ -328,14 +338,96 @@ def test_get_dgl_blocks_hetero():
...
+        {
+            ("n3", "e1", "n1"): ([0, 1, 1], [1, 2, 0]),
+            ("n3", "e2", "n2"): ([0, 1, 2], [1, 0, 2]),
+        }
+    )
+
+    gb_g = gb.from_dglgraph(hg, is_homogeneous=False)
+
+    train_set = gb.HeteroItemSet(
+        {"n1:e1:n2": gb.ItemSet(torch.LongTensor([[0, 1]]), names="seeds")}
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/test_minibatch.py:387]: this etype does not exist

**Review by Rhett-Ying — APPROVED** [review]

**Comment by Rhett-Ying** [comment]
@dgl-bot

**Pull request merged**